test/integration/backward_compatible/serial/CompactRestartTest.js

```diff
@@ -98,4 +101,28 @@ describe('CompactRestartTest', function() {
         const values = await map.valuesWithPredicate(Predicates.sql('INT32 == 42'));
         values.size().should.be.eq(1);
     });
+
+    it('should throw IllegalStateError exception, because of schema could not be replicated in the cluster', async function() {
```

Comment:
I think the test is ok, though we should put it in another file. `test/integration/backward_compatible/parallel/serialization/compact/CompactTest.js` should be better. Don't forget to skip the test if the client version is older than 5.2.0 because we added this behavior in 5.2.0.